tutorials/three-js-tool-tutorial.md

```diff
@@ -0,0 +1,371 @@
+  "modelView matrix" which represents where an object is relative to how the camera sees it.
+
+With that context, the parameters of this function should make a bit of sense. Every frame, as we move the phone
+camera around, this tool is notified of its modelView matrix – how it should be rendered relative to the camera's
+current position. We also get the projection matrix every frame, but we only need to store it once because it never
+changes.
+
+### Step 9.
+
+```javascript
+// 9. only set the projection matrix for the camera 1 time, since it stays the same
+if (!isProjectionMatrixSet && projectionMatrix.length > 0) {
+  setMatrixFromArray(camera.projectionMatrix, projectionMatrix);
+  isProjectionMatrixSet = true;
+}
+```
+
+The first step inside this function is to set the `camera.projectionMatrix` to the projection matrix we get from the
+Spatial Toolbox. This'll make the three.js camera consistent with the way the rest of the Spatial Toolbox is
+rendering. It only needs to happen once, so we wrap it in an if-statement.
+
+### Step 10.
+
+```javascript
+// 10. Every frame, set the position of the containerObj to the modelViewMatrix
+if (isProjectionMatrixSet) {
+  setMatrixFromArray(containerObj.matrix, modelViewMatrix);
+  mesh.rotation.z += 0.01; // make it spin
+  renderer.render(scene, camera);
+}
+```
+
+This step happens every frame after we successfully set the projection matrix. Here, we set the `containerObj.matrix`
+to the modelView matrix from the Spatial Toolbox. This positions all the content of the three.js scene to be viewed
+from the phone's current position.
+
+`mesh.rotation.z += 0.01` is an optional step, which animates the cube so that it slowly rotates in place. This is to
+point out a good spot in the code to add any continuously updating animations or behavior. You can speed it up, stop
+it, scale it up or down, add rotations in other directions, etc.
+
+Calling `renderer.render(scene, camera)` actually draws the resulting pixels onto the screen.
+
+### Step 11.
+
+```javascript
+// 11. This is just a helper function to set a three.js matrix using an array
+function setMatrixFromArray(matrix, array) {
+  matrix.set( array[0], array[4], array[8], array[12],
+              array[1], array[5], array[9], array[13],
+              array[2], array[6], array[10], array[14],
+              array[3], array[7], array[11], array[15]);
+}
+```
+
+You can just copy-and-paste this into your code. Trying to do `containerObj.matrix = modelViewMatrix` doesn't work,
+so instead we have this helper function to use the three.js matrix `set` method to assign a new value to a matrix
+using the data in the format provided by the Spatial Toolbox APIs.
+
+### Step 12.
+
+```
+// 12. Make sure the 3d renderer always has the right aspect ratio of the screen
+window.addEventListener('resize', function() {
+  rendererWidth = window.innerWidth;
```

Comment:
Before these were swapped with a comment about landscape mode, might be an issue?

Reply:
so this is the tricky part that tripped me up the first time (maybe I should clarify in the documentation). for some reason `screen.height` gives the width (at least in landscape mode) but `window.innerWidth` gives the width, not height 😖 . But I can't use window.innerWidth in the beginning because the iframe window starts out 220px by 220px and gets resized to fullscreen after the setFullScreenOn API call.